apps/zk-explorer/components/RunCircuitServiceModalTrigger.tsx

```diff
@@ -10,27 +10,85 @@ import {
   Typography,
 } from '@webb-tools/webb-ui-components';
 import { WEBB_DOCS_URL } from '@webb-tools/webb-ui-components/constants';
-import { FC, useState } from 'react';
+import assert from 'assert';
+import { usePathname, useRouter } from 'next/navigation';
+import { FC, useCallback, useState } from 'react';
 import { RadioCard } from './RadioCard';
 
-export const RunCircuitServiceModalTrigger: FC = () => {
+export type RunCircuitServiceModalTriggerProps = {
+  owner: string;
+  repositoryName: string;
+
+  /**
+   * The unique path of the circuit file under the repository.
+   * This helps identify the circuit file for which the service is being run.
+   *
+   * @remarks
+   * This is the path of the circuit file relative to the repository root.
+   *
+   * If omitted, it is assumed that the user is viewing a file that is not
+   * a circuit file, and thus cannot choose to run a service. The `Run Service`
+   * button will be disabled in this case.
+   *
+   * @example
+   * `apps/zk-explorer/circuits/poseidon.circom`
+   */
+  circuitFilePath?: string;
+};
+
+export const RunCircuitServiceModalTrigger: FC<
+  RunCircuitServiceModalTriggerProps
+> = ({ circuitFilePath }) => {
+  enum RadioItem {
+    TrustedSetupService = 'trusted-setup-service',
+    ProofGenerationService = 'proof-generation-service',
+  }
+
```

Comment:
I suggest placing the decoration outside of the component to prevent it from being re-declared with every render.